### PR TITLE
mariadb: provide mysql-client and add tests

### PR DESCRIPTION
**Pull request by Dentrax**
Provide `mysql-client` on `mariadb-client` package as it's the same package.

### CODE REVIEW

File: mariadb-11.5.yaml [hunk 1]
```diff
@@ -187,6 +187,7 @@ subpackages:
     dependencies:
       provides:
         - mariadb-client=${{package.full-version}}
+        - mysql-client=${{package.full-version}}
```
Comment (mamccorm) [inline mariadb-11.5.yaml:190]: Hey @Dentrax, any more context on this one? Did we remove a duplicate mysql-client package and add this provides instead?

Comment (mamccorm) [reply mariadb-11.5.yaml:190]: Could we add a code comment as well to explain why this provides two separate names?

File: mariadb-11.5.yaml [hunk 2]
```diff
@@ -232,3 +233,42 @@ update:
...
+        mysql --version
+        mariadb --version
+        mariadbd --version
+        mysqladmin --version
+        mariadb-admin --version
+    - name: Test mariadbd start and stop
+      uses: test/daemon-check-output
+      with:
+        setup: |
+          useradd $DBUSER
```
Comment (developer-guy) [inline mariadb-11.5.yaml:261]: if this is something required maybe we could use post-install scriplet for mariadb-client? does that make sense?

Comment (Dentrax) [reply mariadb-11.5.yaml:261]: Actually its just for testing, we shouldn't need this in the image since there will be  `accts` set.

**Review by xnox — CHANGES_REQUESTED** [review]
this will break all of our mysql packaging.

**Pull request closed without merging**